client/lib/cart-values/cart-items.js

```diff
@@ -683,6 +684,25 @@ export function fillGoogleAppsRegistrationData( cart, registrationData ) {
 	);
 }
 
+const getDomainPartFromEmail = emailAddress => {
+	emailAddress = emailAddress || '';
+	return emailAddress
+		.toLowerCase()
+		.trim()
+		.replace( /.*@([^@]+)$/, '$1' );
+};
```

Comment:
Hi @olaseni! Nice to meet you. I noticed here that we're applying some unexpected transforms to the email address. I'm not sure what that last `replace` is attempting to do - it seems to acknowledge that multiple `@`s can exist but if it does so, they are in the wrong position, as they would have to fall to the left of the final `@` and also be quoted.

It's good to be pragmatic about the complexity of email addresses, but some times it's easy to be more robust. In this case that might be true. Also, is there a particular reason to force the lower case?

```js
function emailDomain( email ) {
  if ( 'string' !== typeof email || -1 === email.indexOf( '@' ) ) {
    // there's no safe default that makes sense to me
    return null;
  }
  const parts = email.split( '@' );
  const lastPart = parts[ parts.length - 1 ]; // can be multiple @s before domain, but none after split
  return lastPart.replace( />$/, '' ); // could be "Bob <bob@example.com>"
}
```

Anyway, mine is probably also silly. I just figured that when it's easy to be more spec-compliant then it's worth popping in. Saw the PR in my notifications 🤷‍♂

Reply:
Thanks for your observations! As I think about it now I realise that forcing the lower case may be too early, and can be handled later.

The fact that email can also be specified as `Bob <bob@example.com>` would also be considered . 

However the `replace` does exactly what you specify in a different way i.e. it use a capture group in the regex to get the domain after the last `@` character.

I hope I'm picking up what you are putting down.

Reply:
> However the replace does exactly what you specify in a different way i.e. it uses a regex capture group to grab the string after the last @ character.

Very good! I missed the `$` in the regex when I looked. Might be worth adding a simple explanatory comment…

```js
	.replace( /.*@([^@]+)$/, '$1' ); // domain is everything after last `@`
```